src/Rules/Preg/RegexPatternRule.php

```diff
@@ -0,0 +1,119 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Preg;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+
+class RegexPatternRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	/**
+	 * @param FuncCall $node
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @return string[]
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$patterns = $this->extractPatterns($node, $scope);
+
+		$errors = [];
+		foreach ($patterns as $pattern) {
+			if ($this->validatePattern($pattern)) {
+				continue;
+			}
+
+			$errors[] = sprintf('The regex pattern \'%s\' is not valid.', $pattern);
+		}
+
+		return $errors;
+	}
+
+	/**
+	 * @param FuncCall $functionCall
+	 * @param Scope $scope
+	 * @return string[]
+	 */
+	private function extractPatterns(FuncCall $functionCall, Scope $scope): array
+	{
+		if (!isset($functionCall->args[0])) {
+			return [];
+		}
+		$patternNode = $functionCall->args[0]->value;
+		$patternType = $scope->getType($patternNode);
```

Comment:
Move `$patternType = ...` below the Name condition to save some resources.

Reply:
You could also first check if a related function is called and stop right after that.